### PR TITLE
LPS-184639,LPS-184640

### PR DESCRIPTION
**Pull request by ces-locpham**
Hi team,
Please help me review this PR.

Motivation
=======
In this PR, I solve 2 tasks: 
- [LPS-184639](https://issues.liferay.com/browse/LPS-184639) [BE] Add new field to DLFolder entity
- [LPS-184640](https://issues.liferay.com/browse/LPS-184640) [BE] Update the checkbox value to database when creating new Folder

Proposed Solution
========
For the first task, I added a new column with the name "allowedPropagation" for saving the propagation permission setting.
For the second task, I updated the allowedPropagation property in DLFolderLocalServiceImpl. I did not add more arguments in addFolder function and updateFolder function because this cause a lot of change in the Liferay code. Therefore, I used get allowedPropagation property from service context(3140cece8bbc7e8fba68b61e5c18bfd67907babb))

### CODE REVIEW

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java [hunk 1]
```diff
@@ -801,6 +805,18 @@ public void unlockFolder(long folderId, String lockUuid)
 		LockManagerUtil.unlock(DLFolder.class.getName(), folderId);
 	}
 
+	@Override
+	public void updateAllowedPropagation(
+			long folderId, Boolean allowedPropagation)
```
Comment (ces-luannguyen) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java:810]: I think "boolean" (instead "Boolean") would be better. What do you think?

**Comment by ces-locpham** [comment]
I fixed it @ces-luannguyen . Could you recheck it?

**Review by ces-luannguyen — APPROVED** [review]

**Comment by ces-luannguyen** [comment]
Hi @tototrinh 
This PR has passed my review. Please help double check.

Thanks,
Luan Nguyen

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java [hunk 2]
```diff
@@ -801,6 +805,18 @@ public void unlockFolder(long folderId, String lockUuid)
...
 	}
 
+	@Override
+	public void updateAllowedPropagation(
+			long folderId, boolean allowedPropagation)
+		throws PortalException {
+
+		DLFolder dlFolder = dlFolderPersistence.fetchByPrimaryKey(folderId);
+
+		dlFolder.setAllowedPropagation(allowedPropagation);
```
Comment (tototrinh) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java:815]: I think should check null before setting.

**Review by ces-totrinh — COMMENTED** [review]
Hi @locpham97 @ces-luannguyen,
I have some comments, please take a look. Thanks.

**Comment by ces-totrinh** [comment]
@locpham97, LGTM. Thanks.

**Review by ces-totrinh — APPROVED** [review]

**Comment by ces-totrinh** [comment]
Hi @hhtank, this PR passed my review. Please help take a look. Thanks.

**Comment by ces-locpham** [comment]
The technical draft: https://docs.google.com/document/d/1N2IAiUTPHIwPfWSjf3QWE_HWfIRtU9MOBWP8zFWZL5o/edit?usp=sharing

**Comment by ces-locpham** [comment]
I updated for solving the conflict

File: portal-impl/src/com/liferay/portlet/documentlibrary/service.xml [hunk 3]
```diff
@@ -385,6 +385,7 @@
 		<column name="defaultFileEntryTypeId" type="long" />
 		<column name="hidden" type="boolean" />
 		<column name="restrictionType" type="int" />
+		<column name="allowedPropagation" type="boolean" />
```
Comment (hhtank) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service.xml:388]: I'm quite concerned about the term `allowedPropagation` here. Let me try to explain the idea:
- We're proposing a new term in this feature. It should tell other developers what it is about. Maybe, i think we're not English native speaking so we just try to put a term that is understandable to us internally. But eventually we need to let others understand this new term, especially when this term will be proposed to BChan ;-) i believe we should try to make it more self-explained.
- Reading through the ACs, we sound to add something to store the idea of "permission propagation" (if the folder is a parent folder) and also "permission inheritance" (if the folder is a child). More or less, we may think of a term that can represent in some way both ideas of "propagation" and "inheritance". I'm still not knowing of any English word that can do the job though, maybe we need to ask for some advice from another native English speaker. The only candidate comes to my mind could be something like "permissionPropagation" (or "permissionPropagationEnabled" if boolean value needs to be an adjective).

**Review by hhtank — CHANGES_REQUESTED** [review]
We may consider another term that makes more sense to the developer community.

**Comment by hhtank** [comment]
@tototrinh Just FYI, when you're good to forward the PR, i believe it should go to https://github.com/liferay-staging/liferay-portal

File: portal-impl/src/com/liferay/portlet/documentlibrary/service.xml (hunk 3 above)
Comment (locpham97) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service.xml:388]: Hi @hhtank, 
Thanks for your comments, I will make it more self-explained.
I think it is better to change to the "permissionPropagationEnabled"  fields.
Thanks,
Loc

**Comment by ces-locpham** [comment]
Hi @tototrinh @hhtank ,
I updated the field . Could you recheck it again ?
Thanks,
Loc

**Comment by hhtank** [comment]
@locpham97 Please reword the commit desc accordingly as well. Thanks,

image

**Comment by ces-locpham** [comment]
> @locpham97 Please reword the commit desc accordingly as well. Thanks,
> 
> image

Thanks for your comments @hhtank ,
I updated it

**Comment by hhtank** [comment]
@locpham97 Thanks for the updates! It looks good to me know.

cc @tototrinh

**Comment by ces-totrinh** [comment]
Thanks @hhtank @locpham97
It was merged and waits for the remaining task to complete to be sent to the Product team.

**Pull request merged**